### DISCUSSION TITLE
Dynamic import in node.js app doesn't work: MODULE_NOT_FOUND

### DISCUSSION THREAD

**Discussion by vyshkov** [webpack/webpack#18571] [Q&A]
I am trying to import a js file at runtime:

```
const migrationsDir = path.resolve(__dirname, '../migrations/tenanted');
import(path.join(migrationsDir, '1681831306977-initial-schema.js'))
  .then(() => {
    console.log('done');
  })
  .catch((e) => console.error(e));
```
The file path ('1681831306977-initial-schema.js') is correct, however, it doesn't work if I compile a project with webpack:
```
Error: Cannot find module 'path\1681831306977-initial-schema.js'
    at ...some_path\lambda.js:157949:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'MODULE_NOT_FOUND'
}
```

Webpack compiled the code to be:
```
const migrationsDir = path__WEBPACK_IMPORTED_MODULE_1___default().resolve(__dirname, '../migrations/tenanted');
__webpack_require__("./src/database/migrations lazy recursive")(path__WEBPACK_IMPORTED_MODULE_1___default().join(migrationsDir, '1681831306977-initial-schema.js')).then(()=>{
    console.log('done');
}).catch((e)=>console.error(e));
```

the problem, is that webpack_require("./src/database/migrations lazy recursive") returns `webpackEmptyAsyncContext.`

Is it possible to import an external JS file if I run a node.js app?
I couldn't find a solution on other platforms.

**Comment by alexander-akait** [comment]
Because `import(path.join(migrationsDir, '1681831306977-initial-schema.js'))` is not statically analizable, we don't know what is `path.join(...)`, there are solutions
 - use `require("./migrations/tenanted/1681831306977-initial-schema.js")`
 - use `__non_webpack_require()__`
 - use enable `/* webpackIgnore: true */` comment for commonjs modules and prebuild map of dynamic loaded files
 - replace/include code after webpack build
 
 Feel free to feedback
